app/grants/views.py

```diff
@@ -1487,21 +1520,35 @@ def grant_new(request):
             new_grant(grant, profile)
             add_grant_to_active_clrs(grant)
 
-            # project_pk = request.POST.get('project_pk', None)
-            # if project_pk:
-            #     HackathonProject.objects.filter(pk=project_pk).update(grant_obj=grant)
-
             return JsonResponse({
                 'success': True,
                 'url': grant.url,
             })
 
     project = None
-    # project_id = request.GET.get('project_id', None)
-    # if project_id is not None:
-    #     hackathon_project = HackathonProject.objects.filter(pk=project_id).nocache().first()
-    #     if request.user.profile in hackathon_project.profiles.all():
-    #         project = hackathon_project
+    project_id = request.GET.get('project_id', None)
+    if project_id is not None:
+        hackathon_project = HackathonProject.objects.filter(pk=project_id).nocache().first()
+        if hackathon_project is not None:
+            if hackathon_project.grant_obj is None:
+                if request.user.profile in hackathon_project.profiles.all():
```

Comment:
same optimization here

Reply:
@sanchaymittal could you take a look at @danlipert request ?